lib/input-unmarshaller.js

```diff
@@ -8,21 +8,24 @@ const {
     unmarshaller,
 } = require("./content-negotiation");
 
-const convertToRiffMessage = (payload, headers) => {
-    const messageBuilder = Message.builder();
-    for (const headerName in headers) {
-        messageBuilder.addHeader(headerName, headers[headerName]);
-    }
-    return messageBuilder.payload(payload).build();
-};
-
 module.exports = class InputUnmarshaller extends Transform {
-    constructor(argumentTransformer) {
+    constructor(expectedInputIndex) {
```

Comment:
why do we care about the input index? I don't see anything that varies depending on the input stream? We could make InputUnmarshaller a singleton